(startLine=357 endLine=375 srcPath=/home/sonia/NewExperiment/guavaFilter/00697/guava-tests/test/com/google/common/util/concurrent/FuturesTest.java)
  public void testTransform_getThrowsRuntimeException() throws Exception {
    // This is an evil way to accomplish this, but it works currently.
    AbstractFuture<Object> input =
        new AbstractFuture<Object>() {
          @Override
          public Object get() {
            throw new MyRuntimeException();
          }
        };

    ListenableFuture<Object> output = Futures.transform(input, identity());
    input.set("foo");
    try {
      output.get();
      fail();
    } catch (ExecutionException expected) {
      assertThat(expected.getCause()).isInstanceOf(MyRuntimeException.class);
    }
  }

